code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+            TD_SettingsTable.tdDataStruc{entryNumber_TD} = TDsettings;
+            
+            inStream_TD = 0;
+            streamStopCounter_TD = streamStopCounter_TD + 1;
+            entryNumber_TD = entryNumber_TD + 1;
+            
+        end
+        
+        % POWER DOMAIN
+        if inStream_Power && ~currentSettings.StreamState.PowerDomainStreamEnabled
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            Power_SettingsTable.action{entryNumber_Power} = sprintf('Stop Stream Power %d',streamStopCounter_Power);
+            Power_SettingsTable.recNum(entryNumber_Power) = streamStopCounter_Power;
+            Power_SettingsTable.time{entryNumber_Power} = HostUnixTime;
+            
+            % Fill in most recent power domain settings
+            Power_SettingsTable.powerBands{entryNumber_Power} = powerChannels;
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+            Power_SettingsTable.TDsampleRates{entryNumber_Power} = currentTDsampleRate;
+            Power_SettingsTable.fftConfig(entryNumber_Power) = fftConfig;
+            
+            inStream_Power = 0;
+            streamStopCounter_Power = streamStopCounter_Power + 1;
+            entryNumber_Power = entryNumber_Power + 1;
+            
+        end
+        
+        % FFT
+        if inStream_FFT && ~currentSettings.StreamState.FftStreamEnabled
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            FFT_SettingsTable.action{entryNumber_FFT} = sprintf('Stop Stream FFT %d',streamStopCounter_FFT);
+            FFT_SettingsTable.recNum(entryNumber_FFT) = streamStopCounter_FFT;
+            FFT_SettingsTable.time{entryNumber_FFT} = HostUnixTime;
+            
+            % Fill in most recent FFT settings
+            FFT_SettingsTable.fftConfig(entryNumber_FFT) = fftConfig;
+            
+            % Get sample rate for each TD channel; all TD channels have
+            % same Fs (or is listed as NaN)
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+            FFT_SettingsTable.TDsampleRates{entryNumber_FFT} = currentTDsampleRate;
+            
+            inStream_FFT = 0;
+            streamStopCounter_FFT = streamStopCounter_FFT + 1;
+            entryNumber_FFT = entryNumber_FFT + 1;
+        end
+    end
+    
+    % Option 2: Check if sense has been turned off
+    if isfield(currentSettings,'SenseState')
+        % TIME DOMAIN
+        if inStream_TD && isfield(currentSettings.SenseState,'state')
+            senseState = dec2bin(currentSettings.SenseState.state,4);
```

Comment:
```suggestion
            
```

Reply:
Moved the check of senseState to outside the loops dependent on whether streaming for TD, Power, or FFT. Updated with f6e367e